### ISSUE TITLE
Not Working On BigSur

### ISSUE THREAD

**Issue by EllisDeclan** [DanTheMan827/ios-app-signer#166]
![Screenshot 2020-11-13 at 10 06 26 PM](https://user-images.githubusercontent.com/43008151/99148128-109eb900-26ac-11eb-85a1-71b7be87567f.png)
![Screenshot 2020-11-13 at 10 06 13 PM](https://user-images.githubusercontent.com/43008151/99148129-14324000-26ac-11eb-8238-157d8ee751ef.png)

**Comment by lenisko** [comment]
same for me

**Comment by MrBenFTW** [comment]
Re create your developer certificates in Xcode.

**Comment by lenisko** [comment]
Didn't helped @MrBenFTW

**Comment by MrBenFTW** [comment]
What happens when you let it try and fix itself

**Comment by lenisko** [comment]
![2020-11-21_22-13-01-vmplayer](https://user-images.githubusercontent.com/10072920/99887700-bd49df00-2c46-11eb-906b-fe8c7dd65e7c.png)

It's set as trusted btw

**Comment by FilemonCZ** [comment]
i have the same problem. is there any guidance how to fix or we need to wait for new update? thanks all!

**Comment by kkirby** [comment]
I'm on Big Sur and it works without any issue for me. I didn't have to update anything or swap out profiles or certificates.

**Comment by Schmouly** [comment]
it used to work for me for a while and I'm now encountering the same issue.

**Comment by FilemonCZ** [comment]
kkirby and Schmouly - do you have xcode installed? if i remove xcode it doesnt work. with xcode all works :-(

**Comment by Schmouly** [comment]
yes I have xcode installed. it stops working since the last update of xcode 12.2.
I'm trying to uninstall and reinstall it from scratch to see if it helps the matter.

**Comment by FilemonCZ** [comment]
i have 12.2 xcode and it works.. anyway the idea was to unistall xcode completly and use only apple configurator 2. I guess we need to wait for official answer here

**Comment by Schmouly** [comment]
yes i think so... 🤷🏻‍♂️

**Comment by meyakovenkoj** [comment]
I had the same problem too. In my case the solution was to simply remove the duplicate certificate in Keychain

**Comment by alex-enchi** [comment]
I've checked with `xcode-select -p` to where it points and fixed with  `sudo xcode-select -s /Applications/Xcode.app/Contents/Developer` path to the currently installed xcode.
Worked for me

**Comment by pozineg** [comment]
thanks alex-enchi, worked perfectly

**Comment by lgariv** [comment]
Just spent a few hours on this one. I think I found the fix, but I'm not sure

It required a few things for me to do (some of them might not be necessary, but I'm writing that just in case):
* Have Xcode installed as well as Xcode command line tools (`xcode-select --install`)
* Make sure `xcode-select -p` output contains the directory of your Xcode.app. if it isn't, change in (using `sudo xcode-build -r`)
* Open Keychain Access, select "System", select the "Certificates" tab, and delete those that contain "Apple Developer" (it's ok if there is no such certification) by right clicking on it.
* In Xcode, if you're already logged in to your Apple ID, go to preferences -> accounts, select your Apple ID, click on "manage certificates" and remove all of them.
* Close iOS-app-signer (if opened) and launch again, then use as usual. It should now be working.

**Comment by DanTheMan827** [comment]
@lgariv This isn't a bug with iOS App Signer, it's just the fact that on every OS update, you need to re-install the command line tools.

I suppose I could add a check in that would prompt the user to install the command line tools if they're not present.

This specific issue also includes multiple problems, one is a error caused by certificates in the keychain, the other is caused by xcode command line tools not being installed

**Comment by DanTheMan827** [comment]
@alex-enchi it's definitely not possible to use this without Xcode installed, it uses the command line code signing tools provided _by_ Xcode

**Comment by SigniOS** [comment]
Zsign technical support:
Apple signature tool supporting ios15 and ios16
Support the removal of dylib dynamic library
Completely free, permanently free!
https://urlzs.com/44zCx
